MarkBind/markbind
# Custom timezone for built-in timestamp

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Enhancement to an existing feature


Fixes #1072 



**What is the rationale for this request?**
Allows custom time zones to be used to generate the `timestamp`. Region-specific websites have a use for this feature, as it doesn't always make sense to refer to UTC/GMT time.

**What changes did you make? (Give an overview)**
* Add support for `"timezone"` and `"locale"` params in `site.json`.
* The 2 params above default to `UTC` and `en-GB`. 

**Is there anything you'd like reviewers to focus on?**
N/A

**Testing instructions:**
1. Add the following to `docs/site.json` -
    ```js
    {
    ...
    "timeZone": "Asia/Singapore",
    "locale": "en-US",
    ...
    }
    ```
1. Scroll down and observe the footer. It should show the local Singapore time in the format - 
  `"<Day (short)>, <Month (short)> <Date>, <Year (long)>, <Time> <AM/PM> <Time Zone>"`
  e.g. `"Generated by MarkBind 2.11.0 on Sun, Mar 1, 2020, 10:14:15 PM GMT+8"` 


**Proposed commit message: (wrap lines at 72 characters)**
Support custom timezones for built-in timestamp variable
